archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/context/ContextNode.java

```diff
@@ -347,6 +347,10 @@ private ContextNode resolve(String[] segments, BiFunction<ContextNode, String, C
                 node = root;
             } else if (PARENT_REF.equals(segment)) {
                 node = i == 0 ? parent : node.parent;
+                if (node == null) {
```

Comment:
Can you add a new test in ContextNodeTest that illustrate the use-case that breaks without this change.